physics/module_mp_thompson.F90

```diff
@@ -2188,7 +2214,7 @@ subroutine mp_thompson (qv1d, qc1d, qi1d, qr1d, qs1d, qg1d, ni1d,    &
             endif
             nc(k) = MIN( DBLE(Nt_c_max), ccg(1,nu_c)*ocg2(nu_c)*rc(k)   &
                   / am_r*lamc**bm_r)
-            if (.NOT. is_aerosol_aware) nc(k) = Nt_c
+            if (.NOT. (is_aerosol_aware .or. merra2_aerosol_aware)) nc(k) = Nt_c
```

Comment:
We just follow the previous code for ltaerosol. If you want to delete it please create a separate PR to clean the code in the future.

Reply:
It seems unnecessary to me do this in its own PR.  I'm asking for @grantfirl to weigh in for whether this is OK to do as part of this PR.  It just takes more time later to deal with a set of four one-liners.

Reply:
I just did a regression test by deleting those lines. It changed baseline results.

Reply:
If you look at my original line of code, Nt_c is divided by air density and these lines do not divide by air density.  Perhaps that is the cause.  I kinda put the burden of proof that these lines are needed on someone at EMC.  I believe I coded things correctly to work without this addition.  I went back and checked using Git's "Blame" feature and all I can say is that it was ported 4 years ago by Dom H. and he must have decided to introduce those lines.  They should be unnecessary.

Reply:
To me, it looks like `nc1d` is certainly initialized previously, but it also looks like the preceding code has the possibility to modify `nc` from its initial value. I interpret this line as being protection against modification of `nc` when not using aerosol awareness. This begs the question of why isn't this IF-statement placed higher up if we're just going to reset it down here, but perhaps values of the other variables need a temporarily modified `nc` to calculate, e.g. `nu_c`,`lamc`, etc before `nc` is forced back to its initial value? I'm certainly not an expert in this code and @gthompsnWRF is. My opinion, given @gthompsnWRF expertise, is to make the changes he suggests prior to merging rather than a follow-up UNLESS all of the UFS regression testing has already been finished, no other changes based on Greg's review will change the results, AND it is time-sensitive to get this in. Otherwise, follow-up PRs just kinda add more work over getting the code "right" the first time.

Reply:
I was also going to mention Greg's point that this code has been there since Dom added it to CCPP. Either this line existed from the antecedent codebase he was using (of which I don't know the provenance) or he found it necessary to add for some reason.

Reply:
All the calculations for ```nu_c``` and ```lamc``` should be internally consistent.  If not, then a bug was made somehow.  I reviewed my WRF code and I've never had a line resetting such as these IF-statements.  I suppose I will have to re-check the math, but nu_c is calculated from ```Nt_c```, then ```lamc``` is calculated and so the computations should be "circular" basically.  The division/multipication by rho could be something that got done as an afterthought and introduced an error.

Reply:
In the same IF block, the nc(k) = nc1d(k)*rho(k) again. Maybe there is a reason to not to divide by rho. 
Here is the code:
if (qc1d(k) .gt. R1) then
            no_micro = .false.
            rc(k) = qc1d(k)*rho(k)
            **nc(k) = MAX(2., MIN(nc1d(k)*rho(k), Nt_c_max))**
            .........
            nc(k) = MIN( DBLE(Nt_c_max), ccg(1,nu_c)*ocg2(nu_c)*rc(k)   &
                  / am_r*lamc**bm_r)
            if (.NOT. (is_aerosol_aware .or. merra2_aerosol_aware)) nc(k) = Nt_c
else
............

Reply:
I just checked WRFv4.3.2, this line "if (.NOT. is_aerosol_aware) nc(k) = Nt_c" is in WRF too. So I will leave it as is.  If this is a potential problem, we could create an issue in this fork in the future.

Reply:
Please refer to the following line in the latest WRF model: 
https://github.com/wrf-model/WRF/blob/master/phys/module_mp_thompson.F#L1730

Reply:
This should not be in WRF.  I don't know who added it, but I have been on a much older version for 2-3 years and don't have it in my code - for the obvious reason that I do not believe it to be necessary and only SLOWS the code down by doing more IF-tests.  Too many cooks in the kitchen in my scheme.